Sources/SwiftGRPC/Core/Channel.swift

```diff
@@ -108,11 +129,20 @@ public class Channel {
   /// - Parameter host: the gRPC host name for the call. If unspecified, defaults to the Client host
   /// - Parameter timeout: a timeout value in seconds
   /// - Returns: a Call object that can be used to perform the request
-  public func makeCall(_ method: String, host: String = "", timeout: TimeInterval? = nil) -> Call {
-    let host = host.isEmpty ? self.host : host
-    let timeout = timeout ?? self.timeout
-    let underlyingCall = cgrpc_channel_create_call(underlyingChannel, method, host, timeout)!
-    return Call(underlyingCall: underlyingCall, owned: true, completionQueue: completionQueue)
+  public func makeCall(_ method: String, host: String? = nil, timeout: TimeInterval? = nil) throws -> Call {
+    guard (self.mutex.synchronize { !self.hasBeenShutdown }) else {
+      throw Error.alreadyShutdown
+    }
+
+    guard let underlyingCall = cgrpc_channel_create_call(
```

Comment:
Potential race condition: the channel might have been shut down in between the check above and this call here. You could avoid that by moving the `cgrpc_channel_create_call` into the synchronized closure. (At that point, I would probably just put essentially the entire method's contents into one big sync block, should still be fast enough.)

Reply:
Yep, good call

Reply:
Moved to using a `lock()` and `unlock()` here since it makes throwing/returning simpler